### DISCUSSION TITLE
What is difference between Feature Flag and Configuration?

### DISCUSSION THREAD

**Discussion by joebowbeer** [open-feature/community#315] [General]
I recently learned about OpenFeature by watching Alex Jones's presentation [Building a Kubernetes Experimentation Platform](https://youtu.be/hlvoJhMx2HU)

One question I have is about the difference between _feature flag_ and _configuration_.

According to [one source](https://posthog.com/product-engineers/feature-flags-vs-configuration), feature flag is for the most part an extension of configuration:

> The big difference compared to configurations is feature flags’ expanded functionality

In a [side-by-side comparison](https://posthog.com/product-engineers/feature-flags-vs-configuration#which-should-you-choose), Feature Flag extends Configuration in every category except Versioning.

However, I don't see a comparison between Feature Flag and Configuration in the OpenFeature docs, and I think it would be useful to draw the comparison, if it exists.

Is there anything configuration-wise that OpenFeature considers to be out of scope?

**Comment by liran2000** [comment]
I also found it interesting.
There are some feature flag services which offer feature flags together with "Remote Config". The implementation is very similar, or almost the same sometimes.

\* Regarding "OpenConfig", it is used for networking in a different context, I would remove that from the title.

**Comment by beeme1mr** [comment]
Hey @joebowbeer, I think the article you linked provides a nice comparison. While feature flagging and remote configuration management are very similar from a technical, they have different goals in mind. OpenFeature is focused on feature flagging but could be used for configuration management if the provider supports it and you understand the limitations. Some limitations include no special handling of secrets and potential impact on startup time while the provider syncs with an external service.

Thanks for bringing this up. I'll talk to the governance board to work on an official stance on this topic. It's something we've talked about in the past but the outcome was never captured in the docs.
